src/backend/aspen/api/views/phylo_runs.py

```diff
@@ -269,3 +271,28 @@ async def delete_run(
     await db.delete(item)
     await db.commit()
     return PhyloRunDeleteResponse(id=item_db_id)
+
+
+@router.put("/{item_id}")
+async def update_phylo_tree_and_run(
+    item_id: int,
+    phylo_run_update_request: PhyloRunUpdateRequest,
+    db: AsyncSession = Depends(get_db),
+    user: User = Depends(get_auth_user),
+):
+    # get phylo_run and check that user has permission to update PhyloRun/PhyloTree
+    phylo_run = await get_editable_phylo_run_by_id(db, item_id, user)
+
+    async with db as session:
+        # update phylorun name
+        phylo_run.name = phylo_run_update_request.name
+
+        # if there are any associated PhyloTrees update those names as well:
+        if phylo_run.outputs:
+            # there should only be one phylotree associated with the phylorun
+            phylo_tree = phylo_run.outputs[0]
```

Comment:
It's theoretically possible for a run to have multiple outputs, can we iterate over the outputs looking for one with a PhyloTree type?

Reply:
yes! when would a run have multiple outputs though? I'm just curious

Reply:
Right now this won't happen! It's just future-proof'ing

Reply:
ok i updated it